Core/Src/dynamic_controls/security_architecture.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * security_architecture.c
+ *
+ *  Created on: 2 de mar de 2023
+ *      Author: caius
+ */
+
+#include "dynamic_controls/security_architecture.h"
+
+#include "CAN/general_can_data_manager.h"
+#include "datalogging/datalog_handler.h"
+#include "sensors/sensor_data_processing.h"
+#include "util/global_instances.h"
+#include "util/util.h"
+
+static float speed_mov_avg;
+static float IMU_long_accel_mov_avg;
+static BRAKE_STATUS_t bse_active;
+
+static bool imu_bse_error_status;
+static bool imu_speed_error_status;
+
+void cross_validation(void* argument) {
+    UNUSED(argument);
+
+    for (;;) {
+        ECU_ENABLE_BREAKPOINT_DEBUG();
+
+        osThreadFlagsWait(DYNAMIC_CONTROL_THREAD_FLAG, osFlagsWaitAny, osWaitForever);
```

Comment:
Posso estar interpretando a biblioteca de RTOS de maneira incorreta. Mas entendo que essa thread só será executada uma vez, quando os controles dinâmicos forem ativados na tarefa dynamic_controls_choice.c, que por sua vez só acontece quando o botão de controles é pressionado. Se esse for o caso mesmo, a média móvel não será calculada de maneira correta @FelTell @caiussouza 

```
        osThreadFlagsWait(DYNAMIC_CONTROL_THREAD_FLAG, osFlagsNoClear, osWaitForever);
```

Reply:
Na verdade o erro aqui é outro, a `DYNAMIC_CONTROL_THREAD_FLAG` não é uma thread flag, e sim uma event flag, esse erro é só no nome da flag e no código aqui, o resto tá ok. E você tá certo em falar que precisa do `osFlagsNoClear`.

Reply:
Na verdade o erro aqui é outro, a `DYNAMIC_CONTROL_THREAD_FLAG` não é uma thread flag, e sim uma event flag, esse erro é só no nome da flag e no código aqui, o resto tá ok. E você tá certo em falar que precisa do `osFlagsNoClear`. 

Corrigi no #97 os nomes. Favor dar merge com a branch (ou a develop se o PR for aprovado) e fazer as alterações necessárias.

Reply:
Mas levo outra reflexão, por que rodar essa task apenas quando controles dinâmicos tiver ativado? Não seria melhor rodar o tempo todo e ter o dado de funcionamento para datalogging o tempo todo?

Reply:
Ótima sugestão. Acredito que podemos ampliar o funcionamento dela para além dos controles dinâmicos e até mesmo melhorá-la, adicionando outro parâmetro a ser validado junto. Porém, como essa feature ainda está 100% teórica (pq precisamos do NK em funcionamento pra testar), acho melhor a gnt ver se ela pelo menos cumpre o objetivo principal, se a média realmente é calculada e consegue evitar outliers e se sai do modo de controles dinâmicos se houver erro. Após isso, a gnt pode confiar nela e aumentar o escopo de atuação para todo o tempo em que o carro estiver ligado

Reply:
Não vejo como deixar ela funcionar o tempo todo aumenta o trabalho. Única diferença seria que ela ia rodar o tempo e não precisaria de esperar pela flag de controles dinâmicos. Com isso seria possível monitorar a váriavel sem ter que ficar acionando controles dinâmicos e tals. Sem controles dinâmicos ela não ia fazer nada, apenas setar a flag.

Reply:
Entendi. Vou fazer a adequação do código aqui